### PR TITLE
General code cleanup and dead code elimination

### PR DESCRIPTION
**Pull request by dominic-mulligan-arm**
A pass to reduce the number of compile-time warnings produced by the project, remove deprecated functions that are no longer required due to our new-found embrace of WASI, and so on.

- [x] Remove all functionality marked as deprecated, related to the old programming model and provisioning process.
- [x] Remove all functions that are not used when compiling for any of our targets.
- [x] Remove redundant imports.
- [x] Code health boost by running `cargo clippy` on project components.

(I was originally going to perform a `cargo fmt` pass as well.  Given the size of this changeset, and the sensitivity of some of the changes, I'll defer a `cargo fmt` pass to another pull request.)

### CODE REVIEW

**Comment by dominic-mulligan-arm** [comment]
Whilst this is ready for review, it's best to hold off reviewing this until the Rouille changes are merged and I've rebased this on top of `main`.

**Comment by dominic-mulligan-arm** [comment]
Can somebody take a look at this?

**Comment by dominic-mulligan-arm** [comment]
@gbryant-arm can you review this?

File: veracruz-client/src/veracruz_client.rs
Comment (gbryant-arm) [inline veracruz-client/src/veracruz_client.rs:449]: Is skipping line between blocks something that clippy recommends?

File: veracruz-server-test/src/main.rs
Comment (gbryant-arm) [inline veracruz-server-test/src/main.rs:704]: Should we have a global clippy configuration instead of using attributes? Cf. https://github.com/rust-lang/rust-clippy/blob/master/README.md#configuration

File: veracruz-test/src/main.rs
Comment (gbryant-arm) [inline veracruz-test/src/main.rs:448]: Typo on "satify"

File: runtime-manager/src/runtime_manager_sgx.rs
Comment (dominic-mulligan-arm) [reply runtime-manager/src/runtime_manager_sgx.rs:53]: Clippy seems to strongly prefer `unsafe` functions over `unsafe` blocks.  To be honest, in hindsight I think that this is a mistake, and we should have a dedicated Clippy configuration, as you suggest below, tailored to the project.  I will change this back.

File: sgx-root-enclave/src/lib.rs
Comment (dominic-mulligan-arm) [reply sgx-root-enclave/src/lib.rs:58]: No, once a function is marked `unsafe` then any unsafe code contained within it need not have an `unsafe` block.

File: veracruz-client/src/veracruz_client.rs
Comment (dominic-mulligan-arm) [reply veracruz-client/src/veracruz_client.rs:449]: No, that's something I prefer when reading code.

File: veracruz-server-test/src/main.rs
Comment (dominic-mulligan-arm) [reply veracruz-server-test/src/main.rs:704]: Yes, I think this is a good idea.

File: transport-protocol/src/custom.rs
Comment (ShaleXIONG) [inline transport-protocol/src/custom.rs:18]: If there is a CI problem in sgx, this line might need to stay.

**Review by ShaleXIONG — APPROVED** [review]
`runtime-manager/runtime_manager_enclave` should not be checked in?

File: transport-protocol/src/custom.rs
Comment (dominic-mulligan-arm) [reply transport-protocol/src/custom.rs:18]: The SGX build seems to complete, and (from memory) all of the SGX tests passed, too, though I will double check.

**Comment by dominic-mulligan-arm** [comment]
> `runtime-manager/runtime_manager_enclave` should not be checked in?

Removed.

**Comment by gbryant-arm** [comment]
Regarding the unsuccessful check:
```
23 | pub fn platform_getrandom(buffer: &mut [u8]) -> result::Result {
```

 `result::Result` now embeds a type. Returning `result::Result<()>` should fix it. Rebase error?

**Pull request closed without merging**
